src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/service/NotificationService.kt

```diff
@@ -37,30 +37,49 @@ private const val CRN = "crn"
-    val emailTo = HashSet(allocateCase.emailTo ?: emptySet())
-    emailTo.add(allocationDemandDetails.staff.email!!)
-    if (allocateCase.sendEmailCopyToAllocatingOfficer) emailTo.add(allocationDemandDetails.allocatingStaff.email)
-    MDC.put(REFERENCE_ID, emailReferenceId)
-    MDC.put(CRN, caseDetails.crn)
-    log.info("Email request sent to Notify for crn: ${caseDetails.crn} with reference ID: $emailReferenceId")
-    MDC.remove(REFERENCE_ID)
-    MDC.remove(CRN)
-    return emailTo.map { email -> addRecipientTo400Response(email) { notificationClient.sendEmail(allocationTemplateId, email, parameters, emailReferenceId) } }
+    val laoCase = allocateCase.laoCase
+    if (!laoCase) {
+      val parameters = mapOf(
+        "officer_name" to allocationDemandDetails.staff.name.getCombinedName(),
+        "induction_statement" to mapInductionAppointment(allocationDemandDetails.initialAppointment, caseDetails.type),
+        "requirements" to mapRequirements(allocationDemandDetails.activeRequirements),
+      ).plus(getRiskParameters(notifyData.riskSummary, notifyData.riskPredictors, allocationDemandDetails.ogrs))
+        .plus(getConvictionParameters(allocationDemandDetails))
+        .plus(getPersonOnProbationParameters(allocationDemandDetails.name.getCombinedName(), allocateCase))
+        .plus(getLoggedInUserParameters(allocationDemandDetails.allocatingStaff))
+      val emailTo = HashSet(allocateCase.emailTo ?: emptySet())
+      emailTo.add(allocationDemandDetails.staff.email!!)
+      if (allocateCase.sendEmailCopyToAllocatingOfficer) emailTo.add(allocationDemandDetails.allocatingStaff.email)
+      MDC.put(REFERENCE_ID, emailReferenceId)
+      MDC.put(CRN, caseDetails.crn)
+      log.info("Email request sent to Notify for crn: ${caseDetails.crn} with reference ID: $emailReferenceId")
+      MDC.remove(REFERENCE_ID)
+      MDC.remove(CRN)
+      return emailTo.map { email -> addRecipientTo400Response(email) { notificationClient.sendEmail(allocationTemplateId, email, parameters, emailReferenceId) } }
+    } else {
+      val parameters = mapOf(
+        "officer_name" to allocationDemandDetails.staff.name.getCombinedName(),
+      ).plus(getLoggedInUserParameters(allocationDemandDetails.allocatingStaff))
+        .plus(CRN to allocationDemandDetails.crn)
+      val emailTo = HashSet(allocateCase.emailTo ?: emptySet())
+      emailTo.add(allocationDemandDetails.staff.email!!)
+      if (allocateCase.sendEmailCopyToAllocatingOfficer) emailTo.add(allocationDemandDetails.allocatingStaff.email)
+      MDC.put(REFERENCE_ID, emailReferenceId)
+      MDC.put(CRN, caseDetails.crn)
+      log.info("Email request sent to Notify for crn: ${caseDetails.crn} with reference ID: $emailReferenceId")
+      MDC.remove(REFERENCE_ID)
+      MDC.remove(CRN)
+      return emailTo.map { email -> addRecipientTo400Response(email) { notificationClient.sendEmail(allocationTemplateLAOId, email, parameters, emailReferenceId) } }
+    }
```

Comment:
There is a lot of duplication in the two clauses of the if statement. All that really changes is the parameters map and email template ID.  I would prefer the duplication moved out and parameters and return in the if clause

Reply:
agreed.. sorted